core/model/modx/modx.class.php

```diff
@@ -675,7 +675,7 @@ public function setDebug($debug= true) {
                 parent :: setDebug(false);
             }
             else {
-                $debug = (is_int($debug) ? $debug : defined($debug) ? intval(constant($debug)) : 0);
+                $debug = ((is_int($debug) ? $debug : defined($debug)) ? intval(constant($debug)) : 0);
```

Comment:
Sorry, but the braces are set wrong in your PR. The have to be placed around the second part: 

```
(defined($debug) ? intval(constant($debug)) : 0)
```

Maybe your check would have the same result, but the logic behind is somehow wrong.

Reply:
https://stackoverflow.com/questions/61432488/php-error-unparenthesized-a-b-c-d-e-is-deprecated-use-either-a

Reply:
I see what you are saying and will see how I can update the code!!